lib/Doctrine/DBAL/Logging/DebugStack.php

```diff
@@ -12,7 +12,7 @@ class DebugStack implements SQLLogger
     /**
      * Executed SQL queries.
      *
-     * @var mixed[][]
+     * @var mixed[]|mixed[][]
```

Comment:
What is the problem being fixed? Doesn't `mixed[]` include `mixed[][]`?

Reply:
Different people/tools think different. We need to be more clear about this https://scrutinizer-ci.com/g/doctrine/DoctrineBundle/inspections/5c5ab904-a71a-4a2c-8cfe-2aefd2b4363a/issues/files/Tests/ProfilerTest.php?status=new&orderField=path&order=asc&honorSelectedPaths=0

Reply:
:thinking: I think a proper fix for this error would be:

```suggestion
     * @var array<int,array<string,string>>
```

If you use `mixed[]|mixed[][]`, you're only really sure it's a `mixed[]`

Reply:
> It seems like `array(array('sql' => 'SE...l, 'executionMS' => 1))` of type `array<integer,array<stri...onMS\":\"integer\"}>"}>` is incompatible with the declared type `array<integer,array<integer,*>>` of property `$queries`.

It looks like a bug in Scrutinizer. Let's remove the matching definitions and compare the difference: `array<string>` is incompatible with `array<integer,*>`. The type `string` is definitely compatible with `*` while I'm not sure what the omitted key type means (I thought it implies int).

Could we just replace `mixed[][]` with `array<int,array<string,mixed>>`? It's more specific and may make Scrutinizer happier.

I cannot think of a case when a value of the top-level array is not an array which necessitates adding `mixed[]`.